### PR TITLE
investment-team: split ideation into DesignAgent + DesignReviewAgent + CodeSynthesisAgent

### PR DESCRIPTION
**Pull request by brandonkindred**
## Summary

Splits the Strategy Lab's legacy single-call ideation step into three narrowly-scoped agents and wraps them in a bounded design ↔ design-review loop:

- **`DesignAgent`** — spec only, no `strategy_code`. Has `run` (fresh design from priors + brief) and `revise` (mutates the spec to address a critique).
- **`DesignReviewAgent`** — critique only, no code, no spec mutation. Returns `SpecCritique { ready, issues, rationale, readiness_findings }`. Falls closed on any LLM transport / parse fault.
- **`CodeSynthesisAgent`** — produces Python from a frozen, review-approved spec. Invoked only when the deterministic compiler cannot (`requires_custom_code=True` or `CompilerError`).

The orchestrator's `_run_design_loop` iterates `DesignAgent → SpecReadinessGate → DesignReviewAgent → DesignAgent.revise` up to `MAX_DESIGN_REVIEW_ROUNDS` (default 5, env-overridable via `STRATEGY_LAB_DESIGN_REVIEW_ROUNDS`). Loop exhaustion short-circuits with `status="failed: design_not_ready"`; code-synthesis failure short-circuits with `status="failed: code_synthesis"` — the synthesis loop is never entered against an un-converged spec.

Once a phase exits, downstream phases cannot mutate its outputs: the designer never sees the code, the synthesis agent never rewrites the spec.

## What changed

- `StrategyLabRecord` gains optional `design_rounds: int` and `critiques: list[dict]` for the new audit trail. Populated from both the happy path and the short-circuit path.
- `_run_pre_synthesis_phase` no longer runs `SpecReadinessGate` with `phase="design"` — the design loop is now the sole caller of that phase.
- Phase events renamed: `"ideating"` → `"designing"` (sub-phases `started` / `round_completed` / `ready` / `aborted` / `loopback`) plus new `"design_review"` events. **SSE consumers / UI phase catalog will need updating.**
- `_extract_json` factored out of the old ideation module into `_parse_helpers.extract_json_object` so all three new agents share one tested implementation.
- `IdeationAgent` + `ideation_system.md` removed. `graphs/ideation_swarm.py` re-wired to the new agent surface (filename retained for back-compat; deeper graph rework deferred).

## Tests

Five new test files (40+ cases) plus migrations of every existing test that touched `orch.ideation_agent`:

- `test_strategy_lab_design_agent.py` — run returns spec without code, strips stray `strategy_code`, raises on prose rules, revise renders the critique + prior critiques into the prompt.
- `test_strategy_lab_design_review_agent.py` — ready/not-ready paths, readiness findings persisted on the critique, fail-closed on parse / transport errors, tolerant coercion of mild schema drift.
- `test_strategy_lab_code_synthesis_agent.py` — code returned, markdown fence stripped, spec immutable, transport / empty-response failures raise `CodeSynthesisError`.
- `test_strategy_lab_design_loop.py` — round-1 pass, N-rounds-then-pass, never-ready → short-circuit with persisted critiques, readiness-critical skips reviewer, compile fallback to synthesis agent, code-synthesis failure short-circuit, env override floors to 1.
- `test_strategy_lab_parse_helpers.py` — corner cases for `extract_json_object`.

Coverage on new code: `design.py` 100%, `code_synthesis.py` 100%, `design_review.py` 98%. Full investment-team suite: **1598 passed, 14 skipped**.

## Test plan

- [x] `ruff check .` clean
- [x] `pytest agents/investment_team/tests/` — 1598 passed / 14 skipped
- [x] New env var documented in `CLAUDE.md`
- [ ] SSE consumer / UI phase catalog updated to listen for `"designing"` + `"design_review"` (out of scope here — flag for follow-up)

## Risks / non-goals

- Phase-event rename (`"ideating"` → `"designing"`) is a downstream-visible contract change for any SSE consumer.
- `SpecImplementabilityError` reentry semantics are unchanged here.
- Deeper `graphs/ideation_swarm.py` rework (filename rename, graph reshape) deferred.
- `strategy_ideation_agent.py` (the top-level Investment Team agent) is unchanged — only the inner Strategy Lab `IdeationAgent` is removed.

https://claude.ai/code/session_011Rsgq4bqTrhwBrPZps3aHh

---
_Generated by [Claude Code](https://claude.ai/code/session_011Rsgq4bqTrhwBrPZps3aHh)_

### CODE REVIEW

File: backend/agents/investment_team/strategy_lab/agents/design_review.py [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+
+def _coerce_critique(parsed: Dict[str, Any], readiness_findings: List[str]) -> SpecCritique:
+    """Convert a parsed LLM JSON dict into a :class:`SpecCritique`.
+
+    Tolerant of mild schema drift so a single off-spec issue doesn't
+    discard the rest of the review. Issues with unknown ``field`` values
+    are remapped onto a permissive default (`hypothesis`) so the
+    designer still sees the critique text.
+    """
+    ready = bool(parsed.get("ready", False))
```
[the remark this answers is not in the document]
Comment (brandonkindred) [reply backend/agents/investment_team/strategy_lab/agents/design_review.py]: Good catch — fixed in 2475b32. `_coerce_strict_bool` now accepts only real `bool` or the case-insensitive literals `"true"` / `"false"`; anything else (`""`, `"yes"`, `0`, `1`, `None`) defaults to `False`. Added a parametrized regression test over 9 fail-open inputs plus a `"TRUE"` case-insensitive accept test.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: backend/agents/investment_team/strategy_lab/agents/design_review.py [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+                field="hypothesis",
+                severity="warning",
+                description=(rationale or "Reviewer reported not-ready without naming an issue."),
+                suggested_fix="Tighten the hypothesis or rule definitions.",
+            )
+        )
+
+    return SpecCritique(
+        ready=ready,
+        rationale=rationale,
```
[the remark this answers is not in the document]
Comment (brandonkindred) [reply backend/agents/investment_team/strategy_lab/agents/design_review.py:359]: Good catch — fixed in 2475b32. The coercer now demotes `ready` to `False` whenever any non-info issue is present and appends an audit-trail `CritiqueIssue` documenting the demotion (so the lineage records *why* we overrode the reviewer, not just that we did). Info-only issues are still advisory and don't trigger the demotion. Tests cover the critical, warning, and info-only branches.

---
_Generated by [Claude Code](https://claude.ai/code)_
